Shared/Samples/Create KML multi-track/CreateKMLMultiTrackView.swift

```diff
@@ -0,0 +1,185 @@
+                        await model.locationDisplay.dataSource.stop()
+                        
+                        try await model.saveKMLMultiTrack()
+                        multiTrack = try await model.loadKMLMultiTrack()
+                        
+                        statusText = "Saved KML multi-track to 'HikingTracks.kmz'."
+                    case .reset:
+                        model.reset()
+                        multiTrack = nil
+                        await mapViewProxy.setViewpointScale(model.locationDisplay.initialZoomScale)
+                        
+                        fallthrough
+                    case .startNavigation:
+                        try await model.startNavigation()
+                        statusText = "Tap record to capture KML track elements."
+                    }
+                } catch {
+                    self.error = error
+                }
+            }
+            .task {
+                // Monitors the auto pan mode to determine if recenter button should be enabled.
+                for await autoPanMode in model.locationDisplay.$autoPanMode {
+                    recenterIsEnabled = autoPanMode != .navigation
+                }
+            }
+            .errorAlert(presentingError: $error)
+        }
+    }
+}
+
+private extension CreateKMLMultiTrackView {
+    /// A picker for selecting a track from a KML multi-track.
+    struct TrackPicker: View {
+        /// The KML tracks options shown in the picker.
+        private let tracks: [KMLTrack]
+        
+        /// The proxy for setting map view's viewpoint to the selected track's geometry.
+        private let mapViewProxy: MapViewProxy
+        
+        /// The track selected by the picker.
+        @State private var selectedTrack: KMLTrack?
+        
+        init(multiTrack: KMLMultiTrack, mapViewProxy: MapViewProxy) {
+            tracks = multiTrack.tracks
+            self.mapViewProxy = mapViewProxy
+        }
+        
+        var body: some View {
+            Picker("Track", selection: $selectedTrack) {
+                Text("All Tracks")
+                    .tag(nil as KMLTrack?)
+                
+                ForEach(Array(tracks.enumerated()), id: \.offset) { offset, track in
+                    Text("KML Track #\(offset + 1)")
+                        .tag(track)
+                }
+            }
+            .task(id: selectedTrack) {
+                guard let geometry = selectedTrack?.geometry
+                        ?? GeometryEngine.union(of: tracks.map(\.geometry)) else {
+                    return
+                }
+                
+                await mapViewProxy.setViewpointGeometry(geometry.extent, padding: 25)
+            }
+        }
+    }
+    
+    /// An asynchronous action associated with the sample.
+    enum AsyncAction {
+        case startNavigation
+        case recordTrack
+        case saveKMLMultiTrack
+        case reset
+    }
```

Comment:
It is odd to have this all the way down here when it is used by a state property of the view. It would be better to have this closer to where it is used, usually right before the stored property that uses it.